packages/connect-wallet-modal/src/connectButtons/ConnectWC.tsx

```diff
@@ -62,50 +59,44 @@ export const ConnectWC: FC<ConnectWCProps> = (props: ConnectWCProps) => {
       return; // A user was redirected to a wallet mobile app, no need to continue
     }
 
-    setIsConnecting(true);
-
-    try {
-      onBeforeConnect?.();
-      metricsOnClick?.();
-      disconnect?.();
-
-      const onSuccess = () => {
-        onConnect?.();
-        metricsOnConnect?.();
-      };
-
-      if (WCURICondition && WCURIConnectorFn && WCURIRedirectLink) {
-        // BEGIN Handle connection via redirect using WC Pairing URI (without WalletConnect QR modal)
-        // Because of popup blockers, window.open must be called directly from onclick handler
-        redirectionWindow = window.open('', '_blank');
-        redirectionWindow?.document.write(getRedirectionLoadingHTML());
-
-        await connectAsync({ connector: WCURIConnectorFn });
-        onSuccess();
-
-        // Wait for WalletConnect Pairing URI to arrive
-        if (config.state.current) {
-          const activeConnector = config.state.connections.get(
-            config.state.current,
-          )?.connector;
-          if (activeConnector) {
-            const wcUri = await getWalletConnectUri(activeConnector);
-            setRedirectionWindowLocation(WCURIRedirectLink, wcUri);
-          }
+    onBeforeConnect?.();
```

Comment:
Don't need try/catch here more?

Reply:
As I understand, this `try` block was only for the `finally` block and without `catch` block.
We don't need this `finally` block anymore, therefore we can remove the `try` block.